trview.app/RouteWindow.cpp

```diff
@@ -0,0 +1,254 @@
+            on_waypoint_selected(index);
+        });
+        _waypoints = left_panel->add_child(std::move(waypoints));
+        return left_panel;
+    }
+
+    std::unique_ptr<Control> RouteWindow::create_right_panel()
+    {
+        const float panel_width = 270;
+        auto right_panel = std::make_unique<StackPanel>(Point(), Size(panel_width, window().size().height), Colours::ItemDetails, Size(), StackPanel::Direction::Vertical, SizeMode::Manual);
+        right_panel->set_margin(Size(0, 8));
+
+        auto group_box = std::make_unique<GroupBox>(Point(), Size(panel_width, 110), Colours::ItemDetails, Colours::DetailsBorder, L"Waypoint Details");
+
+        auto stats_box = std::make_unique<Listbox>(Point(10, 21), Size(panel_width - 20, 100), Colours::ItemDetails);
+        stats_box->set_show_headers(false);
+        stats_box->set_show_scrollbar(false);
+        stats_box->set_columns(
+            {
+                { Listbox::Column::Type::String, L"Name", 100 },
+                { Listbox::Column::Type::String, L"Value", 150 }
+            });
+        _token_store.add(stats_box->on_item_selected += [&](const auto& item)
+        {
+            switch (_selected_type)
+            {
+            case Waypoint::Type::Entity:
+                on_item_selected(_all_items[_route->waypoint(_selected_index).index()]);
+                break;
+            case Waypoint::Type::Trigger:
+                on_trigger_selected(_all_triggers[_route->waypoint(_selected_index).index()]);
+                break;
+            }
+        });
+        _stats = group_box->add_child(std::move(stats_box));
+        right_panel->add_child(std::move(group_box));
+
+        // Notes area.
+        auto notes_box = std::make_unique<GroupBox>(Point(), Size(panel_width, window().size().height - 110), Colours::Notes, Colours::DetailsBorder, L"Notes");
+
+        auto notes_area = std::make_unique<TextArea>(Point(10, 21), Size(panel_width - 20, notes_box->size().height - 41), Colour(0.2f, 0.2f, 0.2f), Colour(1.0f, 1.0f, 1.0f));
+        _notes_area = notes_box->add_child(std::move(notes_area));
+
+        right_panel->add_child(std::make_unique<ui::Window>(Point(), Size(panel_width, 5), Colours::Notes));
+        right_panel->add_child(std::move(notes_box));
+
+        _token_store.add(_keyboard.on_char += [&](auto character)
+        {
+            _notes_area->handle_char(character);
+        });
+        _token_store.add(_notes_area->on_text_changed += [&](const std::wstring& text)
+        {
+            if (_route)
+            {
+                _route->waypoint(_selected_index).set_notes(text);
+            }
+        });
+
+        return right_panel;
+    }
+
+    Listbox::Item RouteWindow::create_listbox_item(uint32_t index, const Waypoint& waypoint)
+    {
+        std::wstring type = waypoint_type_to_string(waypoint.type());
+        if (waypoint.type() == Waypoint::Type::Entity)
+        {
+            type = _all_items[waypoint.index()].type();
```

Comment:
There needs to be checking for when the item does not exist in the level (this can happen if the route was made for a different level)

Reply:
Added checks for this.